Improved accuracy of character count when editing posts

stevebaranski opened the pull request:
### Description
Fixes #10343, which observed some inaccuracies in the character count reported when editing a post. 

The solution leverages the same approach used to calculate `wordCount`, with the additional step of adding the length of each word.

The PR consists of the following three changes:

1. It moves the computed properties used for word & character count from `AztecPostViewController` to `UITextView+RichTextView`.
1. It tidies up the plural-handling in the localized string.
1. It adds tests that reproduced (and now verify) the original problem, with some handling of additional cases.

#### Additional PRs

- [X] Move `String` handling to `WordPressShared` [[PR](https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/161/)]
- [X] Update `.podspec` for `WordPressShared` [[PR](https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/163/)]
- [X] Update `.podspec` for `WordPressKit` [[PR](https://github.com/wordpress-mobile/WordPressKit-iOS/pull/52)]
- [X] Update `.podspec` for `WordPressAuthenticator` [[PR](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/33)]

### Testing

- Checkout the branch and verify tests pass.
- Create a post as the issue describes and observe that the character & word counts match expectations.

![10343](https://user-images.githubusercontent.com/221062/48030994-97a51800-e107-11e8-8725-2f0f08359417.png)

jklausa commented on WordPress/Classes/ViewRelated/Views/RichTextView/UITextView+RichTextView.swift:
```diff
@@ -1,8 +1,35 @@
-    @objc func frameForTextInRange(_ range: NSRange) -> CGRect {
+
+    /// Returns a count of valid text characters.
+    var characterCount: Int {
```
This might honestly be overkill for this, but I'd love to see this right next to the implementation of `wordCount` in `NSString+Helpers`. It's a PITA to update since it's in WPShared, but I think it'd benefit from being able to see that the two implementations are so similar to each other.

stevebaranski replied:
This is valid feedback - thank you! 

I agree with the notion of moving this to `WordPressShared`; would you object if I moved it to `String+Helpers.swift` and added some clarifying comment referring to `wordCount` in `NSString+Helpers.h/.m`?

jklausa replied:
Sounds good!

stevebaranski commented:
@jklausa I believe this PR is ready for another look. To recap:

- I moved the associated code for `characterCount` to `WordPressShared`, and initiated the cascade of dependency updates this requires.
- I reverted the commit with the `.stringsdict`, and filed an issue to add this support via `paaHJt-c3-p2`.

Would you mind taking a look when you have a free moment?

stevebaranski commented:
Thanks @jklausa!
